jcl/src/java.base/share/classes/java/lang/invoke/MethodHandleResolver.java

```diff
@@ -222,50 +222,37 @@ private static final Object invokeBsm(MethodHandle bsm, Object[] staticArgs) thr
+		VMLangAccess access = VM.getVMLangAccess();
+		Object internalRamClass = access.createInternalRamClass(j9class);
+		Class<?> classObject = getClassFromJ9Class(j9class);
 
-			int bsmIndex = UNSAFE.getShort(bsmData);
-			int bsmArgCount = UNSAFE.getShort(bsmData + BSM_ARGUMENT_COUNT_OFFSET);
-			long bsmArgs = bsmData + BSM_ARGUMENTS_OFFSET;
-			MethodHandle bsm = getCPMethodHandleAt(internalRamClass, bsmIndex);
-			if (null == bsm) {
-				/*[MSG "K05cd", "unable to resolve 'bootstrap_method_ref' in '{0}' at index {1}"]*/
-				throw new NullPointerException(Msg.getString("K05cd", classObject.toString(), bsmIndex)); //$NON-NLS-1$
-			}
-			Object[] staticArgs = new Object[bsmArgCount];
+		MethodType type = MethodTypeHelper.vmResolveFromMethodDescriptorString(methodDescriptor, access.getClassloader(classObject), null);
+		final MethodHandles.Lookup lookup = new MethodHandles.Lookup(classObject);
+		inaccessibleTypeCheck(lookup, type);
 
-			/* Static optional arguments */
-			int bsmTypeArgCount = bsm.type().parameterCount();
-			for (int i = 0; i < bsmArgCount; i++) {
-				staticArgs[i] = getAdditionalBsmArg(access, internalRamClass, classObject, bsm, bsmArgs, bsmTypeArgCount, i);
-			}
+		int bsmIndex = UNSAFE.getShort(bsmData);
+		int bsmArgCount = UNSAFE.getShort(bsmData + BSM_ARGUMENT_COUNT_OFFSET);
+		long bsmArgs = bsmData + BSM_ARGUMENTS_OFFSET;
+		MethodHandle bsm = getCPMethodHandleAt(internalRamClass, bsmIndex);
+		if (null == bsm) {
+			/*[MSG "K05cd", "unable to resolve 'bootstrap_method_ref' in '{0}' at index {1}"]*/
+			throw new NullPointerException(Msg.getString("K05cd", classObject.toString(), bsmIndex)); //$NON-NLS-1$
+		}
 
-			Object[] appendixResult = new Object[1];
+		Object[] staticArgs = new Object[bsmArgCount];
+		/* Static optional arguments */
+		int bsmTypeArgCount = bsm.type().parameterCount();
+		for (int i = 0; i < bsmArgCount; i++) {
+			staticArgs[i] = getAdditionalBsmArg(access, internalRamClass, classObject, bsm, bsmArgs, bsmTypeArgCount, i);
+		}
 
-			/* result[0] stores a MemberName object, which specifies the caller method (generated bytecodes). */
-			result[0] = MethodHandleNatives.linkCallSite(classObject, 0, bsm, name, type, (Object)staticArgs, appendixResult);;
+		Object[] appendixResult = new Object[1];
+		Object[] result = new Object[2];
+		/* result[0] stores a MemberName object, which specifies the caller method (generated bytecodes). */
+		result[0] = MethodHandleNatives.linkCallSite(classObject, 0, bsm, name, type, (Object)staticArgs, appendixResult);;
 
-			/* result[1] stores a MethodHandle object, which is used as the last argument to the caller method. */
-			result[1] = appendixResult[0];
-		} catch (Throwable e) {
-			if (type == null) {
-				result[0] = new BootstrapMethodError(e);
-			} else {
-				result[0] = e;
-			}
-			/* TODO: Currently, result[0] either stores a MemberName or Throwable. This leads to a type check in the interpreter
```

Comment:
Has this todo been addressed or recorded elsewhere?

Reply:
It is recorded in as item number (2) in https://github.com/eclipse/openj9/issues/10913. I have also re-added the comment in the PR.